### ISSUE TITLE
4D mask created under torch.compile when padding_mask is None

### ISSUE THREAD

**Issue by msnliu** [huggingface/transformers#48925]
### Summary
When running models with `torch.compile` and unpadded inputs (`attention_mask=None`), 4D masks are still materialized instead of passing `attn_mask=None` to SDPA. This causes noticeable SDPA slowdown due to unnecessary mask-add operations.

### Root cause
In `src/transformers/masking_utils.py`:
```python
if is_tracing(padding_mask):
    return False
```
Under torch.compile, is_tracing(None) evaluates to True, which prevents skipping mask creation even when padding_mask is None.

### Expected behavior

When attention_mask=None, create_bidirectional_mask (and create_causal_mask) should return None so SDPA runs without a dense mask.

**Comment by 000wahab000** [comment]
if this issue is still open , can i work on it?

**Comment by msnliu** [comment]
> if this issue is still open , can i work on it?

Hi, thanks for the reply. I already had a PR ready for this, but I would like to receive some feedback from transformers' reviewers.
